pkg/operator/starter.go

```diff
@@ -34,9 +42,55 @@ func RunOperator(clientConfig *rest.Config, stopCh <-chan struct{}) error {
 		kubeClient.RbacV1(),
 	)
 
+	ensureOperatorConfigExists(operator.operatorConfigClient, "v3.11.0/kube-scheduler/operator-config.yaml")
+
 	operatorConfigInformers.Start(stopCh)
 	kubeInformersNamespaced.Start(stopCh)
 
 	operator.Run(1, stopCh)
 	return fmt.Errorf("stopped")
 }
+
+func ensureOperatorConfigExists(client operatorsv1alpha1client.KubeSchedulerOperatorConfigsGetter, path string) {
+	v1alpha1Scheme := runtime.NewScheme()
+	v1alpha1.Install(v1alpha1Scheme)
+	v1alpha1Codecs := serializer.NewCodecFactory(v1alpha1Scheme)
+	operatorConfigBytes := v311_00_assets.MustAsset(path)
+	operatorConfigObj, err := runtime.Decode(v1alpha1Codecs.UniversalDecoder(v1alpha1.GroupVersion), operatorConfigBytes)
+	if err != nil {
+		panic(err)
+	}
+	requiredOperatorConfig, ok := operatorConfigObj.(*v1alpha1.KubeSchedulerOperatorConfig)
+	if !ok {
+		panic(fmt.Sprintf("unexpected object in %s: %t", path, operatorConfigObj))
+	}
+
+	hasImageEnvVar := false
+	if imagePullSpecFromEnv := os.Getenv("IMAGE"); len(imagePullSpecFromEnv) > 0 {
+		hasImageEnvVar = true
+		requiredOperatorConfig.Spec.ImagePullSpec = imagePullSpecFromEnv
+	}
+
+	existing, err := client.KubeSchedulerOperatorConfigs().Get(requiredOperatorConfig.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		if _, err := client.KubeSchedulerOperatorConfigs().Create(requiredOperatorConfig); err != nil {
+			panic(err)
```

Comment:
many panics around here

Reply:
yes so far of operator config does not exist, that is fatal, can not proceed.